### ISSUE TITLE
CREATING THE PASSWORD INPUT WIDGET FAILED. ('PasswordInput' object has no            attribute '_on_change')

### ISSUE THREAD

**Issue by kakos-k9** [beeware/toga#1616]
```
import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW


class TESTE(toga.App):

    def startup(self):
        
        # WIDGETS ###############################
        self.password = toga.PasswordInput()
        #self.number = toga.NumberInput()
        self.pushButton = toga.Button('AHHHH')
        ########################################

        # BOX ####################################################
        main_box = toga.Box(style=Pack(direction=COLUMN))
        main_box.add(self.password, self.pushButton)
        #########################################################

        # EVENTS #####################################################
        self.password.on_change = self.funcao_teste
        #self.pushButton.on_press = self.printar
        ##############################################################

        # WINDOW #####################################################
        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()
        ##############################################################

    #def printar(self, widget):
    #    brasil = float(self.number.value)
    #    print(brasil)


    def funcao_teste(self, widget):
        print("BRAZIL")


def main():
    return TESTE()
```
[briefcase.2022_10_14-15_22_24.run.log](https://github.com/beeware/toga/files/9789111/briefcase.2022_10_14-15_22_24.run.log)

**Comment by kakos-k9** [comment]
Using: toga-android==0.3.0.dev35

**Comment by freakboy3742** [comment]
I suspect this issue has been resolved with some recent fixes to TextInput. Can you verify if the problem still exists with the current `main` version?

**Comment by kakos-k9** [comment]
I don't quite understand this "main" of what you are referring to. But in this example I showed I'm using briefcase 0.3.10 with toga0.3.0-dev35

In briefcase version 0.3.9 also using toga0.3.0-dev35 this error did not happen

**Comment by mhsmith** [comment]
```
Caused by: com.chaquo.python.PyException: AttributeError: 'PasswordInput' object has no attribute
           '_on_change'                                                                                                                                                             
       at <python>.toga.widgets.textinput.on_change(textinput.py:118)
       at <python>.toga_android.widgets.textinput.afterTextChanged(textinput.py:22)
       at <python>.chaquopy_java.call(chaquopy_java.pyx:379)
       at
           <python>.chaquopy_java.Java_com_chaquo_python_PyObject_callAttrThrowsNative(chaquopy_java.pyx:351)                                                                       
       at com.chaquo.python.PyObject.callAttrThrowsNative(Native Method)
       at com.chaquo.python.PyObject.callAttrThrows(PyObject.java:232)
       at com.chaquo.python.PyInvocationHandler.invoke(PyInvocationHandler.java:31)
       at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
       at $Proxy10.afterTextChanged(Unknown Source)
       at android.widget.TextView.sendAfterTextChanged(TextView.java:10805)
       at android.widget.TextView.setText(TextView.java:6407)
       at android.widget.TextView.setText(TextView.java:6227)
       at android.widget.EditText.setText(EditText.java:121)
       at android.widget.TextView.setText(TextView.java:6179)
       at android.widget.TextView.setTransformationMethod(TextView.java:2661)
       at android.widget.TextView.setInputType(TextView.java:6658)
       at <python>.java.chaquopy.CQPEnv.check_exception(env.pxi:546)
       at <python>.java.chaquopy.CQPEnv.CallNonvirtualVoidMethodA(env.pxi:242)
       at <python>.java.chaquopy.JavaMethod.call_nonvirtual_method(class.pxi:884)
       at <python>.java.chaquopy.JavaMethod.__call__(class.pxi:795)
       at <python>.java.chaquopy.JavaMethod.__get__.lambda2(class.pxi:775)
       at <python>.toga_android.widgets.passwordinput.create(passwordinput.py:8)
       at <python>.toga_android.widgets.base.__init__(base.py:33)
       at <python>.toga.widgets.passwordinput._create(passwordinput.py:9)
       at <python>.toga.widgets.textinput.__init__(textinput.py:43)
       at <python>.TESTE.app.startup(app.py:21)
       at <python>.toga_android.app.create(app.py:173)
       at <python>.toga_android.app.main_loop(app.py:184)
       at <python>.toga.app.main_loop(app.py:517)
       at <python>.__main__.<module>(__main__.py:4)
```

**Comment by mhsmith** [comment]
Actually the error does happen with Briefcase 0.3.9, but Rubicon simply logs the exception and continues:

```
10-17 13:49:42.262 19548 19548 I Python  : Python app launched & stored in Android Activity class
10-17 13:49:42.431 19548 19548 D Python  : Native invocation 126790074283584 :: beforeTextChanged
10-17 13:49:42.435 19548 19548 D Python  : Native invocation done.
10-17 13:49:42.440 19548 19548 D Python  : Native invocation 126790074283584 :: onTextChanged
10-17 13:49:42.442 19548 19548 D Python  : Native invocation done.
10-17 13:49:42.448 19548 19548 D Python  : Native invocation 126790074283584 :: afterTextChanged
10-17 13:49:42.456 19548 19548 E Python  : Traceback (most recent call last):
10-17 13:49:42.458 19548 19548 E Python  :   File "/data/user/0/com.example.helloworld/files/python/user_code/app_packages/rubicon/java/api.py", line 54, in dispatch
10-17 13:49:42.458 19548 19548 E Python  :     val = getattr(pyinstance, method)(*args)
10-17 13:49:42.460 19548 19548 E Python  :   File "/data/user/0/com.example.helloworld/files/python/user_code/app_packages/toga_android/widgets/textinput.py", line 22, in afterTextCha
10-17 13:49:42.460 19548 19548 E Python  :     if self.interface.on_change:
10-17 13:49:42.461 19548 19548 E Python  :   File "/data/user/0/com.example.helloworld/files/python/user_code/app_packages/toga/widgets/textinput.py", line 118, in on_change
10-17 13:49:42.461 19548 19548 E Python  :     return self._on_change
10-17 13:49:42.462 19548 19548 E Python  : AttributeError: 'PasswordInput' object has no attribute '_on_change'
10-17 13:49:42.463 19548 19548 D Python  : Native invocation done.
```

The error does not happen with Briefcase 0.3.10 and Toga 0.3.0.dev38, so this was probably fixed by #1593 and #1610.
